app/src/main/java/io/homeassistant/companion/android/sensors/SensorWorker.kt

```diff
@@ -43,6 +44,16 @@ class SensorWorker(private val appContext: Context, workerParams: WorkerParamete
             .appComponent((appContext as GraphComponentAccessor).appComponent)
             .build()
             .inject(this)
+
+        // This will cause the sensor to be updated every time the OS broadcasts that a cable was plugged/unplugged.
+        // This should be nearly instantaneous allowing automations to fire immediately when a phone is plugged
+        // in or unplugged.
+        appContext.registerReceiver(
+            ChargingBroadcastReceiver(integrationUseCase), IntentFilter().apply {
+                addAction(Intent.ACTION_POWER_CONNECTED)
+                addAction(Intent.ACTION_POWER_DISCONNECTED)
```

Comment:
Does this also trigger for wireless charging or are those other intents?

Reply:
According to the Android docs it should. I can confirm this with an emulator though.

Reply:
Unfortunately it doesn't look like the emulator actually supports any plug type other than AC which makes testing this tough. However according to the Android docs this approach is sound as any type of power connection should theoretically fire this Intent.